src/condor_starter.V6.1/job_info_communicator.cpp

```diff
@@ -703,8 +703,23 @@ JobInfoCommunicator::initUserPrivWindows( void )
 	bool init_priv_succeeded = true;
 	bool run_as_owner = allowRunAsOwner( false, false );
 
-	if( !name ) {	
-		if ( run_as_owner ) {
+	if (run_as_owner) {
+		char *run_jobs_as;
+		if (job_ad->LookupString(ATTR_OS_USER, &run_jobs_as)) {
```

Comment:
this breaks run-as-owner on Windows for older AP's ?

Reply:
No. The code to lookup Owner and NTDomain is still there as a fallback below. The check of 'name' here is pointless, as it's always NULL. There used to be some VM universe code just above that could set 'name' to a non-NULL value.